### PR TITLE
#577 - API Update Changes and removal of libcloud dependency

### PR DESCRIPTION
**Pull request by rszabo50**
#577

Fixes issues with latest release of the API

### CODE REVIEW

File: saltcloud/libcloudfuncs.py [hunk 1]
```diff
@@ -332,3 +337,19 @@ def conn_has_method(conn, method_name):
         )
     )
     return False
+
+def is_private_subnet(ip):
```
Comment (techhat) [inline saltcloud/libcloudfuncs.py]: Two comments here. First, only functions specific to libcloud belong in this file. Second, this duplicates functionality which already existts in the is_public_ip() function in utils/**init**.py.

File: saltcloud/clouds/joyent.py [hunk 2]
```diff
@@ -139,44 +157,37 @@ def create(vm_):
     Create a single VM from a data dict
     '''
 
+    #pprint(inspect.stack())
```
Comment (techhat) [inline saltcloud/clouds/joyent.py]: Forgotten?

File: saltcloud/clouds/joyent.py [hunk 3]
```diff
@@ -255,74 +281,81 @@ def create(vm_):
     return ret
 
 
-def reboot(name, call=None):
-    '''
-    Reboot a node.
+def create_node(**kwargs):
```
Comment (techhat) [inline saltcloud/clouds/joyent.py:291]: This is one of many functions that doesn't contain docstrings. Each function needs at least a one line description, though a CLI Example is not necessarily required unless usage isn't obvious. With all the new additions lately, that is likely to change (meaning CLI Examples will probably be required soon).

**Comment by rszabo50** [comment]
i will make those changes tomorrow and you can review again

File: saltcloud/libcloudfuncs.py (hunk 1 above)
Comment (rszabo50) [inline saltcloud/libcloudfuncs.py]: agreed, and did not find that function earlier. will change and test tomorrow

File: saltcloud/clouds/joyent.py (hunk 2 above)
Comment (rszabo50) [inline saltcloud/clouds/joyent.py]: needs to be removed, used it to find full stack trace when debugging an earlier issue

**Comment by rszabo50** [comment]
this is ready for review

**Pull request merged**